### PR TITLE
[bugfix][a11y][update] Associate in this call with username for React 16

### PR DESCRIPTION
**Pull request by edwardlee-msft**
# What

Replace useId with a custom communication react id generator to support React 16.

Screenshot 2023-06-20 at 11 11 01
Screenshot 2023-06-20 at 11 11 13
Screenshot 2023-06-20 at 11 12 01

# Why

useId is a new a hook introduced in React 18. Our 1.x.x library needs to support React 16+.
https://skype.visualstudio.com/SPOOL/_workitems/edit/3118296

# How Tested
MacOs CallWithChat sample voiceover

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-composites/src/composites/common/utils.tsx [hunk 1]
```diff
@@ -53,3 +53,12 @@ export const useMinMaxDragPosition = (modalLayerHostId: string, rtl?: boolean):
  * @private
  */
 export const defaultSpokenLanguage = 'en-us';
+
+/**
+ * @private
+ *  * Generate a unique id
+ * TODO: Replace with useId() once React 18 becomes a required dependency.
+ */
+export const generateUniqueId = (): string => {
```
Comment (dmceachernmsft) [inline packages/react-composites/src/composites/common/utils.tsx:62]: are we placing this hook in both files with the hope to replace them some day? should we not put this in the `acs-ui-common` package so we can use these in other places too if need be?

**Review by dmceachernmsft — APPROVED** [review]

File: packages/react-composites/src/composites/common/utils.tsx (hunk 1 above)
Comment (edwardlee-msft) [reply packages/react-composites/src/composites/common/utils.tsx]: Good catch, yeah that's right. I moved it to acs-ui-common now! :D

File: packages/react-composites/src/composites/common/ParticipantContainer.tsx [hunk 2]
```diff
@@ -51,7 +52,7 @@ export const ParticipantListWithHeading = (props: {
   onFetchParticipantMenuItems?: ParticipantMenuItemsCallback;
 }): JSX.Element => {
   const { onFetchAvatarPersonaData, onFetchParticipantMenuItems, title, participantListProps } = props;
-  const subheadingUniqueId = useId();
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/common/ParticipantContainer.tsx:54]: Can you use useId from fluent instead?

Comment (edwardlee-msft) [reply packages/react-composites/src/composites/common/ParticipantContainer.tsx]: Good suggestion! Changed to use just that

**Review by JamesBurnside — APPROVED** [review]

**Pull request merged**